src/libexpr/flake/call-flake.nix

```
@@ -12,7 +12,36 @@ overrides:
 
 let
 
-  lockFile = builtins.fromJSON lockFileStr;
+  origLockFile = builtins.fromJSON lockFileStr;
+
+  # Build an optimized lock file by consolidating equivalent inputs.
+  lockFile =
+    let
+      nodeCacheKey = key: node:
+        let taint = if overrides ? ${key} then { taintedAs = key; } else {};
+        in builtins.toJSON (node.locked or {} // taint);
+
+      inputsCache = builtins.listToAttrs (builtins.attrValues
```

```suggestion
      nodeKeyByDeduplicationKey = builtins.listToAttrs (builtins.attrValues
```

It's a bit unfortunate that "node key" and "deduplication key" are both keys. Can't think of a good rename though.